src/30_Advanced/SeatMap.html

```diff
@@ -0,0 +1,175 @@
+        box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.25);
+        background-color: white;
+        border-radius: 3px;
+      }
+      .opaque {
+        opacity: 0.5;
+        filter: alpha(opacity=50);
+      }
+      .seat-map-svg {
+        box-sizing: border-box;
+        padding-top: 30px;
+      }
+      #pan-zoom {
+        display: initial;
+      }
+      #pan-zoom > svg {
+        left: 0;
+        right: 0;
+        margin: auto;
+      }
+      .list-container {
+        position: relative;
+        height: 50vh;
+      }
+      .ampstart-card {
+        min-width: 100%;
+        padding: 1rem;
+      }
+    </style>
+    <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style>
+    <noscript>
+      <style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style>
+    </noscript>
+
+  </head>
+  <body>
+    <!-- ## Seatmap -->
+    <!-- Here we are using an `amp-state` called `seatmap`.
+      `scale` variable is used to keep track of the scale of the content being zoomed using by `amp-pan-zoom`.
+      `zoom` variable is used to keep track of the seat map is in a zoomed state or not. This variable is used to
+    decide whether to show a + or - button.
+    `selectedSeats` variables keeps a  list of selected seat ids.  -->
+    <amp-state id="seatmap">
+      <script type="application/json">
+        {
+          "scale": 1,
+          "selectedSeats": [],
+          "zoom": 0
+        }
+      </script>
+    </amp-state>
+
+    <!-- You can implement a seat map by using `amp-list`, `amp-bind` and `amp-pan-zoom`.
+    `amp-list` is used to keep the availability of the seatmap up to date.
+    `amp-bind` is used to update the status of the seats: when a seat is selected or unselected we update the color and 
+    we add the seat id to the `seatmap.selectedSeats` variable. 
+    `amp-pan-zoom` is used to make possible to pan & zoom on mobile devices.
+    We are using the `transformEnd` action from [amp-pan-zoom](https://www.ampproject.org/docs/reference/components/amp-pan-zoom)
+    to update the value of transform.scale variable.
+    -->
+    <div>
+     <h1>Seat map</h1>
+     <div class="list-container">
+     <amp-list layout="fill" src="/json/seats.json" items="." single-item noloading>
+      <!--~
+      We display a placeholder for the seatmap by using server-side templating. This is possible because we have
```

Comment:
What does the placeholder look like? Why are we doing it?

I don't understand what you're trying to say with: `This is possible because we have  the information about the layout of the theater.`